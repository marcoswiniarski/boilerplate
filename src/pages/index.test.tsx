import { render, screen } from '@testing-library/react'
import Home from '.'

describe('<Home />', () => {

  it('Deve renderizar o header da pÃ¡gina', () => {
    const { container } = render(<Home />)

    expect(screen.
      getByRole('heading',
        { name: /Boilerplate com NextJS, ReactJS, Typescript e StyledComponents/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Deve renderizar os estilos corretamente', () => {
    const { container } = render(<Home />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#8e44ad',
      'color': '#fff'
    })
  })
})     