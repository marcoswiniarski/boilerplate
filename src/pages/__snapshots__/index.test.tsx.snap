// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Deve renderizar o header da página 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-color: #8e44ad;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 62.5%rem;
  text-align: center;
}

.c2 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.c3 .code {
  color: #000;
  background: #fafafa;
  border-radius: 5px;
  padding: 0.254rem;
  font-family: Menlo,Monaco,Lucida Console,Liberation Mono,DejaVu Sans Mono,Bitstream Vera Sans Mono,Courier New,monospace;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Boilerplate com NextJS, ReactJS, Typescript e StyledComponents
  </h1>
  <h2
    class="c2"
  >
    Sem stress, configurado e pronto para uso. Experimente !
  </h2>
  <p
    class="c3"
  >
    Para começar é so editar o arquivo 
    <code
      class="code"
    >
      pages/index.tsx
    </code>
  </p>
</main>
`;
